### DISCUSSION TITLE
revert commit

### DISCUSSION THREAD

**Discussion by codedi99er** [community/community#27450] [Other Feature Feedback, Questions, & Ideas] [answered]
Can I revert a commit?

**Comment by jhonniel** [comment, marked as answer]
Yes by using 

$ git revert

An 'undo' command, though not a traditional undo operation. Instead of removing the commit, it figures out how to invert the changes in the commit, then appends a new commit with the inverse content. This prevents Git from losing history, which is important for the integrity of your revision history and for reliable collaboration.

**Comment by airtower-luna** [comment]
In case you actually want to change the history (which is generally only advisable as long as you haven't shared/merged it yet), `git rebase -i` is what you need. See: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by jfullstackdev** [comment]
just want to comment on too, 

also if it's done through a pull request you can revert it 

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/reverting-a-pull-request
